libcalico-go/lib/apis/v3/ipam_block.go

```diff
@@ -39,15 +39,46 @@ type IPAMBlock struct {
 
 // IPAMBlockSpec contains the specification for an IPAMBlock resource.
 type IPAMBlockSpec struct {
-	CIDR           string                `json:"cidr"`
-	Affinity       *string               `json:"affinity,omitempty"`
-	StrictAffinity bool                  `json:"strictAffinity"`
-	Allocations    []*int                `json:"allocations"`
-	Unallocated    []int                 `json:"unallocated"`
-	Attributes     []AllocationAttribute `json:"attributes"`
+	// The block's CIDR.
+	CIDR string `json:"cidr"`
 
+	// Affinity of the block, if this block has one. If set, it will be of the form
```

Comment:
Tried to document these since the way the IPAMBlock API isn't obvious for new contributors.